docs/component-testing.md

```diff
@@ -0,0 +1,138 @@
+    search: '',
+    hash: ''
+  }
+}
+```
+
+Let's write a more complex test with a render of the react component:
+
+```typescript
+// src/tests/test.testplane.tsx
+import { useState } from 'react';
+import { render } from '@testing-library/react';
+
+// A simple component with a title and a counter button
+function Component() {
+    const [count, setCount] = useState(0);
+
+    return (
+        <div id="root">
+            <h1>Testplane Component Testing</h1>
+            <button onClick={() => setCount((count) => count + 1)}>
+                count is {count}
+            </button>
+        </div>
+    );
+}
+
+it('should render react button', async ({browser}) => {
+    render(<Component />); // rendering the component on the generated Vite page
+
+    const button = await browser.$("button");
+
+    await button.click();
+    await button.click();
+
+    await expect(button).toHaveText("count is 2");
+});
+```
+
+A fully working examples can be found [here](../examples/component-testing/).
+
+> ⚠️ Currently, there are the following restrictions:
+> - only components written in React in files `.jsx` and `.tsx` are supported. Ability to write tests in `.js` files will be implemented soon. We will also support the Vue framework in the near future;
+> - there is no access to `currentTest` from `it`, `beforeEach` and `afterEach`. It will appear in the near future;
+> - the [@testplane/global-hook](https://github.com/gemini-testing/testplane-global-hook) plugin is temporarily not supported.
+
+
+### What additional features are supported?
+
+#### Hot Module Replacement (HMR)
+
+[HMR](https://vitejs.dev/guide/api-hmr.html) is supported in Vite. This means that if you change the loaded file, either the component will be remounted, or the page will be completely preloaded. If the component is described in a separate file (i.e. not in the same file as the test), a remount will be performed, but the test will not be restarted. And if you change the test file, the page will be reloaded, which will cause Testplane to interrupt the execution of the current test and start it again. Due to this feature, you can quickly develop components in Vite and write tests for them. It is recommended to use it together with the [REPL mode](./cli.md#repl-mode).
+
+#### Using the browser and expect instances directly in the browser DevTools
+
+Instances of the `browser` and `expect` are available inside the browser in which the test is being performed. It is quite convenient to use it when debugging the test.
+
+#### Logs from the browser console in the terminal
+
+Calling the `log`, `info`, `warn`, `error`, `debug` and `table` commands on the `console` object in the browser causes information to be displayed not only in the browser's DevTools, but also in the terminal from which Testplane was launched. I.e., you can call `console.log` in the test/component and then see the result of it execution in the terminal. This is also handy when debugging the test.
```

Comment:
> and then see the result

"And you will be able to see the result"

> This is also handy

I suggest "This is especially handy"